### DISCUSSION TITLE
[Feature Request] Multiple errors handling

### DISCUSSION THREAD

**Discussion by odyfey** [riverqueue/river#633] [Ideas]
When using the `errros.Join` to return a result from a `Work` function, all errors are recorded as a single string in the database, which can be inconvenient when viewing a large number of errors in RiverUI. Below is an example code:

```go
func (e *Example) Work(_ context.Context, job *river.Job[ExampleArgs]) error {
	var result []error

       ...

	return errors.Join(result...)
}
```

The reason for this behaviour may be use of [ErrorStr](https://github.com/riverqueue/river/blob/master/job_executor.go#L318) method instead of [errors.Unwrap](https://pkg.go.dev/errors#Unwrap) when recording an error in the database.

How do you feel about supporting the recording of a slice of errors in the database and displaying them in RiverUI as a list separated by newlines?

**Comment by bgentry** [comment]
Hmm, so `errors.Unwrap` merely accesses the next nested error, but it still doesn’t address the fact that it would need to be rendered as a string to be stored in the database. Is your proposal to build an array of the strings for each nested error? In most cases I imagine these will be quite redundant since outer errors may render the text of inner errors.

Maybe I’m misunderstanding the proposal but I’m not sure this would work well for the general case. 🤔

**Reply by odyfey** [reply]
> Hmm, so errors.Unwrap merely accesses the next nested error

Not exactly, in the case where the error was created using the `errors.Join` method, `errors.Unwrap` returns an array of errors.

> Is your proposal to build an array of the strings for each nested error?

No, my suggestion is to handle the case where the worker function returns a slice of errors. This is usually useful when we want the worker to iterate over an array of entities, perform certain actions on each iteration, and in case of an error, not interrupt the entire processing cycle.

Here is how the output of multiple errors currently looks in RiverUI:

<img width="1302" alt="image" src="https://github.com/user-attachments/assets/053a93cc-2a9d-445c-8839-5c253c66029d">

My suggestion is to improve the UX by displaying each error on a separate line.

**Comment by nexovec** [comment]
Just curious, would it be bad just encode the errors as `JSONB` to allow for nested errors?

**Reply by bgentry** [reply]
I'm having trouble picturing how that would work in practice. Can you elaborate on what that might actually look like with an example?

**Reply by nexovec** [reply]
I thought it was as easy as unwrapping the nested errors into a tree struct and calling `json.Marshal` on it, instead of storing it as a string array. I must be missing something?
